File: SDKs/googleapis.com/jobs/v4/python/src/sdk/models/shared/completequeryresponse.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import completionresult as shared_completionresult
from ..shared import responsemetadata as shared_responsemetadata


@dataclass_json
@dataclasses.dataclass
class CompleteQueryResponse:
    r"""CompleteQueryResponse
    Response of auto-complete query.
    """
    
    completion_results: Optional[list[shared_completionresult.CompletionResult]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('completionResults') }})
    metadata: Optional[shared_responsemetadata.ResponseMetadata] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('metadata') }})
    

File: SDKs/googleapis.com/jobs/v3p1beta1/java/lib/src/main/java/openapisdk/models/shared/CompleteQueryResponse.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * CompleteQueryResponse
 * Output only. Response of auto-complete query.
**/
public class CompleteQueryResponse {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("completionResults")
    public CompletionResult[] completionResults;
    public CompleteQueryResponse withCompletionResults(CompletionResult[] completionResults) {
        this.completionResults = completionResults;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("metadata")
    public ResponseMetadata metadata;
    public CompleteQueryResponse withMetadata(ResponseMetadata metadata) {
        this.metadata = metadata;
        return this;
    }
}